gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -128,6 +130,31 @@ public String getProp(String key, String def) {
   }
 
   /**
+   * Get the value of a property as a case insensitive {@link java.util.Set} of strings.
+   *
+   * @param key property key
+   * @return value associated with the key as a case insensitive {@link java.util.Set} of strings
+   */
+  public Set<String> getPropAsCaseInsensitiveSet(String key) {
+    Set<String> set = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+    set.addAll(Splitter.on(",").trimResults().omitEmptyStrings().splitToList(getProperty(key)));
```

Comment:
You can use `split` instead of `splitToList` that uncecessarily creates a temporary list.